misc/python/materialize/checks/all_checks/source_tables.py

```diff
@@ -0,0 +1,257 @@
+                """,
+                f"""
+                > CREATE TABLE pg_table_1b FROM SOURCE pg_source_{self.suffix} (REFERENCE "pg_table_1");
+                > CREATE TABLE pg_table_2 FROM SOURCE pg_source_{self.suffix} (REFERENCE "pg_table_2");
+
+                $ postgres-execute connection=postgres://postgres:postgres@postgres
+                INSERT INTO pg_table_1 VALUES (4, 3456);
+                INSERT INTO pg_table_2 VALUES (3000);
+                """,
+            ]
+        ]
+
+    def validate(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+                > SELECT * FROM pg_table_1;
+                1 1234
+                2 0
+                3 2345
+                4 3456
+
+                > SELECT * FROM pg_table_1b;
+                1 1234
+                2 0
+                3 2345
+                4 3456
+
+                > SELECT * FROM pg_table_2;
+                1000
+                2000
+                3000
+
+                > SELECT * FROM pg_table_1_old_syntax;
+                1 1234
+                2 0
+                3 2345
+                4 3456
+           """
+            )
+        )
+
+
+class TableFromMySqlSource(TableFromSourceBase):
+    suffix = "tbl_from_mysql_source"
+
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            self.generic_setup()
+            + dedent(
+                f"""
+                $ mysql-connect name=mysql url=mysql://root@mysql password={MySql.DEFAULT_ROOT_PASSWORD}
+
+                > CREATE SECRET mysqlpass_{self.suffix} AS 'p@ssw0rd';
+
+                > CREATE CONNECTION mysql_conn_{self.suffix} TO MYSQL (
+                    HOST mysql,
+                    USER root,
+                    PASSWORD SECRET mysqlpass_{self.suffix}
+                  )
+
+                $ mysql-execute name=mysql
+                DROP DATABASE IF EXISTS public_{self.suffix};
+                CREATE DATABASE public_{self.suffix};
+                USE public_{self.suffix};
+
+                CREATE TABLE mysql_source_table_1 (a INTEGER, b INTEGER);
+                INSERT INTO mysql_source_table_1 VALUES (1, 1234), (2, 0);
+
+                CREATE TABLE mysql_source_table_2 (a INTEGER);
+                INSERT INTO mysql_source_table_2 VALUES (1000), (2000);
+
+                > CREATE SOURCE mysql_source_{self.suffix} FROM MYSQL CONNECTION mysql_conn_{self.suffix};
+
+                > CREATE SOURCE old_mysql_source_{self.suffix} FROM MYSQL CONNECTION mysql_conn_{self.suffix} FOR TABLES (public_{self.suffix}.mysql_source_table_1 AS mysql_table_1_old_syntax);
+                """
+            )
+        )
+
+    def manipulate(self) -> list[Testdrive]:
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                f"""
+                > CREATE TABLE mysql_table_1 FROM SOURCE mysql_source_{self.suffix} (REFERENCE "public_{self.suffix}"."mysql_source_table_1");
```

Comment:
we could add a text columns option here if we made one of the table columns an `enum` value or `year` value which are the 2 types that are unsupported natively and we've added text-columns support for: https://materialize.com/docs/sql/create-source/mysql/#supported-types